### PR TITLE
Add SPDX identifiers and prettier for contracts

### LINKED ISSUE #1169: Add SPDX license identifiers
We get reams of these warnings during every compilation:

```
Warning: SPDX license identifier not provided in source file. Before publishing, consider adding a comment containing "SPDX-License-Identifier: <SPDX-License>" to each source file. Use "SPDX-License-Identifier: UNLICENSED" for non-open-source code. Please see https://spdx.org for more information.
--> project:/contracts/extensions/votingReputation/VotingReputationStorage.sol
```

Let's add the identifiers and save our logs.

### PR DESCRIPTION
**Pull request by area**
Closes #1169

We talked about this on our call yesterday; this is a very rudimentary attempt to add [`prettier`](https://github.com/prettier-solidity/prettier-plugin-solidity) to the contracts. There's a lot of good here, some bad as well, but opening it up for discussion.

~~(NB Build doesn't work yet due to some of the bad)~~

### CODE REVIEW

**Comment by area** [comment]
Licensing is so fun

* Our files get `GPL-3.0-or-later`, which is pretty straightforward.
* The ENS files originally from the [ENS repository](https://github.com/ensdomains/ens/blob/master/contracts/ENS.sol) are under `BSD-2-Clause`
* Metatransaction files were adapted from [Biconomy](https://github.com/bcnmy/metatx-standard/blob/basic-signature-metatx/src/contracts/BasicMetaTransaction.sol) which at the time were under GPL3.0, and get `GPL-3.0-or-later`. That repository is now under the MIT license, but unsure if anything changed in the files after we used them and before the licence changed.
* Multicall files came from [Uniswap](https://github.com/Uniswap/v3-periphery/blob/main/LICENSE) and are licensed under `GPL-2.0-or-later`.
* [DSToken](https://github.com/dapphub/ds-token) and friends are `GPL-3.0-or-later`.
* The [Patricia Trie](https://github.com/chriseth/patricia-trie/tree/master) implementation is `MIT`
* Migrations.sol from [Truffle](https://github.com/trufflesuite/truffle/blob/develop/LICENSE) is `MIT`.

File: contracts/extensions/votingReputation/VotingReputationStorage.sol [hunk 1]
```diff
@@ -359,52 +343,48 @@ contract VotingReputationStorage is ColonyExtension, BasicMetaTransaction, Votin
...
     assembly {
-              // call contract at address a with input mem[in…(in+insize))
-              //   providing g gas and v wei and output area mem[out…(out+outsize))
-              //   returning 0 on error (eg. out of gas) and 1 on success
+      // call contract at address a with input mem[in…(in+insize))
+      //   providing g gas and v wei and output area mem[out…(out+outsize))
+      //   returning 0 on error (eg. out of gas) and 1 on success
 
-              // call(g,   a,  v, in,                insize,        out, outsize)
+      // call(g,   a,  v, in,                insize,        out, outsize)
```
Comment (kronosapiens) [inline contracts/extensions/votingReputation/VotingReputationStorage.sol:382]: Should add spaces here to line the comment back up with the call

File: contracts/patriciaTree/Bits.sol [hunk 2]
```diff
@@ -33,11 +33,11 @@ library Bits {
     require(self != 0, "colony-patricia-tree-zero-self");
     uint val = self;
     for (uint8 i = 128; i >= 1; i >>= 1) {
-      if (val & (ONE << i) - 1 == 0) { // ignore-swc-101 TODO: this one also only shows up if analysing PatricaTree.sol
+      if (val & ((ONE << i) - 1) == 0) {
+        // ignore-swc-101 TODO: this one also only shows up if analysing PatricaTree.sol
```
Comment (kronosapiens) [inline contracts/patriciaTree/Bits.sol:37]: Does moving the comment to the next line mess up the slither parsing? We may need to move it to the line above and add `ignore-next-line` or something

File: contracts/reputationMiningCycle/ReputationMiningCycleRespond.sol [hunk 3]
```diff
@@ -94,7 +93,7 @@ contract ReputationMiningCycleRespond is ReputationMiningCycleCommon {
   // Note that these values and the mining window size (defined in ReputationMiningCycleCommon)
   // need to be consistent with each other, but are not checked, in order for the decay
   // rate to be as-expected.
-  uint constant DECAY_NUMERATOR =    999679150010889; // 1-hr mining cycle
+  uint constant DECAY_NUMERATOR = 999679150010889; // 1-hr mining cycle
```
Comment (kronosapiens) [inline contracts/reputationMiningCycle/ReputationMiningCycleRespond.sol:96]: I would ignore prettier here

**Review by kronosapiens — COMMENTED** [review]
Overall looks good. My main bit of feedback is that I think we should set the line width to something shorter, either 80 or 120. There are also some new imports which seem redundant to me, but for which there is probably some reason.

I would also be in favor of replacing all instances of `+= 1` with `++`

File: scripts/generate-test-contracts.sh [hunk 4]
```diff
@@ -38,11 +38,4 @@ sed -i.bak "s/contract ReputationMiningCycle/contract UpdatedReputationMiningCyc
...
 sed -i.bak "s/interface IReputationMiningCycle/interface IUpdatedReputationMiningCycle/g" ./contracts/reputationMiningCycle/IUpdatedReputationMiningCycle.sol
-sed -i.bak "s/function resetWindow() public;/function resetWindow() public; function isUpdated() public pure returns(bool);/g" ./contracts/reputationMiningCycle/IUpdatedReputationMiningCycle.sol
-# Modify VotingReputationMisaligned to have the correct version
-
-votingVersion="$(grep 'return [0-9]*;' ./contracts/extensions/votingReputation/VotingReputation.sol | sed 's/    return //' | sed 's/;//')"
-echo "Current Voting contract version is $votingVersion"
-previous_version=$(($votingVersion - 1))
-echo "Updating test contract to $previous_version"
-sed -i.bak "s/return 4/return $previous_version/g" ./contracts/testHelpers/VotingReputationMisaligned.sol
+sed -i.bak "s/function resetWindow() public;/function resetWindow() public; function isUpdated() public pure returns(bool);/g" ./contracts/reputationMiningCycle/IUpdatedReputationMiningCycle.sol
```
[the remark this answers is not in the document]
Comment (area) [reply scripts/generate-test-contracts.sh:41]: We [removed the test](https://github.com/JoinColony/colonyNetwork/pull/1090/files) that uses that file. I actually meant to remove the solidity file as well, but I guess I missed committing it.

I want this test to come back with the new approach, but that's not for this PR.

File: contracts/patriciaTree/Bits.sol (hunk 2 above)
Comment (area) [reply contracts/patriciaTree/Bits.sol:37]: Seemingly not, as the build is green?

File: .prettierrc [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+    "plugins": ["prettier-plugin-solidity"],
+    "overrides": [
+        {
+            "files": "*.sol",
+            "options": {
+                "parser": "solidity-parse",
+                "printWidth": 140,
```
[the remark this answers is not in the document]
Comment (area) [reply .prettierrc:8]: I've gone aggressive here, as this is not a [hard wrap](https://prettier.io/docs/en/options.html#print-width) like other tools.

File: contracts/colonyNetwork/ColonyNetworkExtensions.sol [hunk 6]
```diff
@@ -18,23 +19,19 @@
...
-import "./../metaTxToken/MetaTxToken.sol";
-import "./../common/TokenAuthority.sol";
-
+import {ColonyDataTypes} from "../colony/ColonyDataTypes.sol";
+import {IColonyNetwork} from "../colonyNetwork/IColonyNetwork.sol";
+import {ColonyExtension} from "../extensions/ColonyExtension.sol";
+import {ColonyNetworkStorage} from "./ColonyNetworkStorage.sol";
+import {MetaTxToken} from "./../metaTxToken/MetaTxToken.sol";
+import {TokenAuthority} from "./../common/TokenAuthority.sol";
+import {EtherRouter} from "./../common/EtherRouter.sol";
```
[the remark this answers is not in the document]
Comment (area) [reply contracts/colonyNetwork/ColonyNetworkExtensions.sol:28]: The old syntax implicitly imported everything from a file that was imported. This syntax only imports what is specifically defined in the import command, so we lose those implied imports. We could have imported everything we used from the files we already had, but I thought it better to 'go to the source' for our imports with the intention of making it clearer.

The one place I opted to not do this was where we imported both interfaces and data types, as I thought those were closely linked enough to bring in from the same place where appropriate, but I'd be happy to go explicit for those too.

File: .prettierrc (hunk 5 above)
[the remark this answers is not in the document]
Comment (kronosapiens) [reply .prettierrc]: It still seems as though it's making lines longer than they were currently (a lot of `require` statements which had spanned multiple lines are now one long line).

File: .prettierrc [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        {
+            "files": "*.sol",
+            "options": {
+                "parser": "solidity-parse"
+            }
+        }
+    ],
+    "tabWidth": 2,
+    "printWidth": 80,
+    "trailingComma": "all",
```
Comment (kronosapiens) [inline .prettierrc:13]: I might be inclined to drop this, it seems to mostly add cruft to our javascript

File: contracts/colony/Colony.sol [hunk 8]
```diff
@@ -39,25 +48,45 @@ contract Colony is BasicMetaTransaction, Multicall, ColonyStorage, PatriciaTreeP
...
+  }
+
+  function emitSkillReputationReward(
+    uint256 _skillId,
+    address _user,
+    int256 _amount
+  ) public stoppable auth validGlobalOrLocalSkill(_skillId) {
     require(_amount > 0, "colony-reward-must-be-positive");
-    IColonyNetwork(colonyNetworkAddress).appendReputationUpdateLog(_user, _amount, _skillId);
+    IColonyNetwork(colonyNetworkAddress).appendReputationUpdateLog(
```
Comment (kronosapiens) [inline contracts/colony/Colony.sol:85]: I like 80 better than 140, but it does seem to wrap almost every call... how would you feel about 100 as a target?

File: .prettierrc (hunk 5 above)
[the remark this answers is not in the document]
Comment (kronosapiens) [reply .prettierrc]: I've gone ahead an upped it to 100 and I'm generally happy with the results, which seem more balanced

**Comment by kronosapiens** [comment]
I've re-reviewed this with the line break set to 100 characters and I'm much happier with it.

File: contracts/colonyNetwork/ColonyNetworkExtensions.sol (hunk 6 above)
[the remark this answers is not in the document]
Comment (area) [reply contracts/colonyNetwork/ColonyNetworkExtensions.sol]: I don't think it was happening in either case, but the new, more specific, imports mean that this should be _less_ likely than it was before, unless I'm misunderstanding something? Most of the new imports are to allow type-checking, as well, which is only at compile-time.

[the remark this answers is not in the document]
Comment (kronosapiens) [reply contracts/colonyNetwork/ColonyNetworkExtensions.sol]: My thinking was more around what happens if we explicitly import a contract (such as `IColonyNetwork`) in a contract which has already inherited the same contract (by inheriting `ColonyStorage`), for example. But if it's a non-issue and the compiler avoids double-inclusion then I'm all good.

**Review by kronosapiens — APPROVED** [review]

**Pull request merged**